### DISCUSSION TITLE
Lack of documentation

### DISCUSSION THREAD

**Discussion by flefevre** [open-webui/open-webui#3761] [General]
Open WebUI is useful for me every day.
I am still impressed of so many features.
But I think there is a lack of documentation on those cool features and how to use them .

Do you think your team can dedicate time to write some documentation?

I know it is time consuming and no so fun, but it is really a 'plus' for an open project

**Comment by justinh-rahb** [comment]
No disagreement that docs can _always_ be improved. PRs are always welcomed, and docs PRs especially usually always get merged as long as they're well-composed.

**Comment by JohnUiterwyk** [comment]
Hi,
 just to re-raise this @justinh-rahb, as someone looking to select an open source chat client to start working from and contributing to, i'm impressed by open web ui's capabilities. Thank you for all your hard work. I would love to see growth of this project, especially in enterprise settings.

 The challenge is getting up to speed on the current stack, and how to work with it, is very difficult given the state of the documentation. I would love to contribute but the state of the documentation is barrier to me contributing, especially to the docs themselves. While there is probably much of the core information already written down, just needing reorginiztion, there critical missing information. 
For example, for me to move from a bespoke chat client i built to using Open Web UI, means i would need to convert some existing functionality i have built into Pipes/Functions etc. But there is no adequate programming/library style documentation on how to author these, as in i can't tell what function gets called when and how, and what the values that will be passed in etc, let alone how to test it. There needs to be at the very least documentation of the class structure (with a full example), the required and optional methods, the types and example values of the inputs, and the lifecycle of the invocation. I would gladly write this documentation, but i don't know the information nor do i know how to get it, without picking through the code trying to figure out what is called where and how. The net result of this is that you are creating a barrier to adoption by capable people, who could, once up to speed, be huge contributors. The good documentation will be the single biggest impact to new user adoption and contributor growth, and once you have a good foundation, people can contribute, but in its current state the documentation blocks adoption.

also attn: @tjbck

**Reply by tjbck** [reply]
@JohnUiterwyk Feel free to ping me on our discord! Will answer all your questions there, so I can help you work on the docs!
